### ISSUE TITLE
[Feature Request] "I don't like this image: Skip!"

### ISSUE THREAD

**Issue by bentolor** [neffo/bing-wallpaper-gnome-extension#72]
Thanks for this great addon.

I observe myself sometimes to be not that happy with the current picture of the day and find myself pressing the reload button franticly w/o any results.

Would it be possible to offer a "Skip this" menu option which falls back to the last non-skipped image?

#### Luxurious version
Just came in my mind: Or maybe even a carousel gallery instead the current static image preview to navigate forth & back?

**Comment by neffo** [comment]
This is an interesting idea, I'm always keen to hear how people are using this extension (or would like to use this extension). I'll look into the carousel idea, but certainly a "revert to yesterday" or "skip this/black list this image" option would be a possibility.

**Comment by neffo** [comment]
I've been testing some features in a test branch here: https://github.com/neffo/bing-wallpaper-gnome-extension/tree/multi-wallpaper

At the moment it's just a list of images which can be selected, along with the default mode (always display the current image), and a random mode (change once per hour). (It will have a minimum of 7 images listed, but will keep history of any images saved to disk once this version is installed. Unfortunately, not possible to go back beyond these as the extension doesn't have the historical data even if the images are still on disk. It might be possible to restore some of it, but not a priority atm.)

Future features:
- the carousel with options to remove/blacklist images

It's only been tested on GNOME 3.38 and GNOME 40, and likely contains bugs. Feedback would be appreciated.

To install:

1. Delete, rename or back up this folder: $HOME/.local/share/gnome-shell/extensions/BingWallpaper@ineffable-gmail.com
2. Clone the test branch:
```
git clone -b multi-wallpaper https://github.com/neffo/bing-wallpaper-gnome-extension.git $HOME/.local/share/gnome-shell/extensions/BingWallpaper@ineffable-gmail.com
```
You can update that with these commands (as it's seeing a bit of work atm):
```
cd $HOME/.local/share/gnome-shell/extensions/BingWallpaper@ineffable-gmail.com
git pull
```

![image](https://user-images.githubusercontent.com/5569112/116799879-8a820c00-ab3f-11eb-9aca-96a040bc0f6d.png)

Similar feature requests:
#89
#59

**Comment by bentolor** [comment]
Thanks @neffo for your improvement and extensive description. I installed the version and had i now running for a few days. 

I appreciate your effort to value my user feedback, but I think the feature in the current form is not ready for showtime. At least in my use case.

I'll just give you my UX story:

1. I installed you version
2. Had an annoying "picutre of the day". Found it a little cumbersome but navigated to the setting and changed to a different pictre
3. I forgot about it
4. Several days later, I stumbled over the image and wondered what it is: Opening the status icon reveals nothing than an empty listing. Navigating to the preferences setting reveals nothing only an empty combobox.(regarding image entries)

While I see at least benefits for te _"Most recent image"_ vs. _"Random image"_, option, I do think the select a specific image has currently two flaws:

* The setting inevitably expires after a week and
* In my use case I'm rather about _not this one_ other than _exactly that one_.

So I'll go back and stick to _"Most recent image" for now.

**Comment by neffo** [comment]
It was probably an unfinished and buggy state at the time (it's still not quite ready to merge and publish), but hopefully that works a bit better. We keep a store of the image data from Bing so titles in menus should be populated, and that state is restored on start up.

I've added the next & previous buttons from the Windows app (which I've never used), which works nicely with the in menu thumbnail. I'll add a setting to reset to current image when a new image arrives (which should probably be the default behavior). I hadn't thought through how to handle that to be honest.

A selected image shouldn't expire, unless you delete the images after 7 days (default is to keep). Regarding your use case, I am thinking a blacklist of images could work once I get the image carousel working (for those who use random).

Thanks for your feedback and testing the bleeding edge version.

**Comment by bentolor** [comment]
Yes, that sounds much better and really awesome @neffo !

From my perspective this is already very close to my expectations: I don't think a "blacklist" is really necessary. 
Rather: _"As a user, as soon as I press prev/next buttons to change the current image this, this images stays, _until_ a new day/image arrives and gets displayed."_

Though it might be possible a user want to permanently switch to a specific image, I don't think this is a major use case. More exactly the whole point of this plugin is to get a fresh stream of image everyday. So i think the _static image_ case could be ruled out in the design consideration.

Regarding "expiration" I might have been ambiguous: The desktop image was stable, it was just that the icon menu and the preferences menu no longer displayed any image information nor presented a list of available images (even newer ones). But maybe this has improved already.

I got a little tricked: Obviously in Gnome central you released a new version and this erased my custom version on restart: I restored the development branch again and look forward to test the forth/back button!

Again, thanks for you response & effort!
